source/extensions/filters/http/cache/cache_filter.cc

```diff
@@ -126,6 +123,17 @@ Http::FilterDataStatus CacheFilter::encodeData(Buffer::Instance& data, bool end_
   return Http::FilterDataStatus::Continue;
 }
 
+void CacheFilter::onAboveWriteBufferHighWatermark() { encode_buffer_high_watermark_ = true; }
+
+void CacheFilter::onBelowWriteBufferLowWatermark() {
+  ASSERT(encode_buffer_high_watermark_);
+  encode_buffer_high_watermark_ = false;
+  if (!remaining_ranges_.empty()) {
+    // Cached body fetch was stopped, resume fetching.
```

Comment:
There could be an outstanding call to getBody. If so, calling getBody again will likely blow the buffer. Also, we don't yet know how much data we'll get back from the outstanding call, so we don't know what to request.